File: 944.删列造序.py
#
# @lc app=leetcode.cn id=944 lang=python
#
# [944] 删列造序
#
# 解法1(T60% S38%)：没什么好说的，纯数组遍历的题，也没什么坑，两重循环解决

# @lc code=start
class Solution(object):
    def minDeletionSize(self, strs):
        """
        :type strs: List[str]
        :rtype: int
        """
        res = 0
        for j in range(len(strs[0])):
            for i in range(len(strs)):
                if i == 0: continue
                if strs[i][j] < strs[i-1][j]:
                    res += 1
                    break
        return res
# @lc code=end

File: javascript/944.删列造序.js
/*
 * @lc app=leetcode.cn id=944 lang=javascript
 *
 * [944] 删列造序
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {number}
 */
var minDeletionSize = function(strs) {
    var strs_T = Array.from(new Array(strs[0].length), () => new Array(strs.length));
    for (var i=0; i<strs_T.length; i++) {
        for (var j=0; j<strs_T[0].length; j++) {
            strs_T[i][j] = strs[j][i];
        }
    }
    
    var res = 0;
    for (var i=0; i<strs_T.length; i++) {
        var flag = true;
        for (var j=1; j<strs_T[i].length; j++) {
            if (strs_T[i][j] < strs_T[i][j-1]) {
                flag = false;
                break;
            }
        }
        if (!flag) {
            res++;
        }
    }
    return res;
};
// @lc code=end

